crates/core/src/error.rs

```diff
@@ -1,810 +1,451 @@
+        let new_issue_url = self.new_issue_url.as_ref().unwrap_or(&default_issue_url);
 
-/// [`StdInErrorKind`] describes the errors that can be returned while dealing IO from CLI
-#[derive(Error, Debug, Display)]
-pub enum StdInErrorKind {
-    /// error reading from stdin: `{0:?}`
-    StdInError(#[from] std::io::Error),
-}
+        write!(
+            f,
+            "\n\nIf you think this is an undiscovered bug, please open an issue at: {new_issue_url}"
+        )?;
 
-/// [`ArchiverErrorKind`] describes the errors that can be returned from the archiver
-#[derive(Error, Debug, Display)]
-pub enum ArchiverErrorKind {
-    /// tree stack empty
-    TreeStackEmpty,
-    /// cannot open file
-    OpeningFileFailed,
-    /// option should contain a value, but contained `None`
-    UnpackingTreeTypeOptionalFailed,
-    /// couldn't get size for archive: `{0:?}`
-    CouldNotGetSizeForArchive(#[from] BackendAccessErrorKind),
-    /// couldn't determine size for item in Archiver
-    CouldNotDetermineSize,
-    /// failed to save index: `{0:?}`
-    IndexSavingFailed(#[from] IndexErrorKind),
-    /// failed to save file in backend: `{0:?}`
-    FailedToSaveFileInBackend(#[from] CryptBackendErrorKind),
-    /// finalizing SnapshotSummary failed: `{0:?}`
-    FinalizingSnapshotSummaryFailed(#[from] SnapshotFileErrorKind),
-    /// `{0}`
-    #[error(transparent)]
-    FromPacker(#[from] PackerErrorKind),
-    /// `{0}`
-    #[error(transparent)]
-    FromTree(#[from] TreeErrorKind),
-    /// `{0}`
-    #[error(transparent)]
-    FromConfigFile(#[from] ConfigFileErrorKind),
-    /// `{0}`
-    #[error(transparent)]
-    FromStdIo(#[from] std::io::Error),
-    /// `{0}`
-    #[error(transparent)]
-    FromStripPrefix(#[from] StripPrefixError),
-    /// conversion from `u64` to `usize` failed: `{0:?}`
-    ConversionFromU64ToUsizeFailed(TryFromIntError),
+        if let Some(backtrace) = &self.backtrace {
+            write!(f, "\n\nBacktrace:\n{backtrace:?}")?;
+        }
+
+        Ok(())
+    }
 }
 
-/// [`VfsErrorKind`] describes the errors that can be returned from the Virtual File System
-#[derive(Error, Debug, Display)]
-pub enum VfsErrorKind {
-    /// No directory entries for symlink found: `{0:?}`
-    NoDirectoryEntriesForSymlinkFound(OsString),
-    /// Directory exists as non-virtual directory
-    DirectoryExistsAsNonVirtual,
-    /// Only normal paths allowed
-    OnlyNormalPathsAreAllowed,
-    /// Name `{0:?}`` doesn't exist
-    NameDoesNotExist(OsString),
+// Accessors for anything we do want to expose publicly.
+impl RusticError {
+    /// Creates a new error with the given kind and guidance.
+    pub fn new(kind: ErrorKind, guidance: impl Into<SmolStr>) -> Box<Self> {
+        Box::new(Self {
+            kind,
+            guidance: guidance.into(),
+            context: Cow::default(),
+            source: None,
+            error_code: None,
+            docs_url: None,
```

Comment:
constructor code is "duplicated" several times.
would there be a way to reduce that without cost ? 

`Default`? `builder pattern`? `...spread operator`?

Reply:
Thanks for the reminder, I did that during sketching and actually forgot!